public/utils/utils.tsx

```diff
@@ -63,6 +69,19 @@ export const validateName = (
   }
 };
 
+export const validateCustomResultIndex = (name: string): string | undefined => {
```

Comment:
In future this method can also be used when user tries inputing custom data source index.

Reply:
Currently we allow user to choose concrete index or input index pattern like "abc*". So can't reuse this part directly. We can revisit/refactor this part in future if needed.